elastic/kibana
# [Event Log] Skip setting assets to hidden in serverless

Resolves https://github.com/elastic/kibana/issues/163891.

In this PR, I'm making the event log skip converting existing assets to hidden if it is in a serverless deployment. Given assets that need to be converted to hidden would exist prior to 8.0, there isn't such a case in serverless and we do not have the necessary legacy APIs to retrieve them.

## To verify on Serverless
- Ensure on serverless there are no more warnings as indicated in https://github.com/elastic/kibana/issues/163891.

## To verify on non-Serverless
1. Create a 7.15 deployment (and persisting the ES data so you can upgrade later)
2. Create an alerting rule (with short interval, say 10s)
3. Let the alerting rule run a few times so it generates event logs
4. Upgrade ES to 7.17 (it will fail otherwise)
5. Upgrade your deployment to this PR
6. Check the event log index template, indices and alias for 7.15 to ensure they got converted to hidden
  - `curl -XGET http://elastic:changeme@localhost:9200/_template/.kibana-event-log-7.15.3-template`
  - `curl -XGET http://elastic:changeme@localhost:9200/.kibana-event-log-7.15.3-000001`
  - `curl -XGET http://elastic:changeme@localhost:9200/_alias/.kibana-event-log-7.15.3`
